### DISCUSSION TITLE
'window.location.replace' vs 'router.replace'

### DISCUSSION THREAD

**Discussion by Dann1y** [vercel/next.js#38400] [Help] [answered]
What is the difference between `window.location.replace` vs `router.replace` in next.js ?

If I want to visit another page(totally another website) and redirect to current page, which one is more fit in this case?

```javascript
<Component handleSomthing={() => router.replace('redirect url')} />
``` 
or
```javascript
<Component handleSomthing={() => window.location.replace('redirect url')} />
```

**Comment by icyJoseph** [comment, marked as answer]
`push`, `replace`, `back`, and the `Link` are all used for SPA like transitions, meaning that the site doesn't unload and reload to show the new page. 

`back` is a bit special, because the router can listen to it, and update its internal knowledge of the current history state.

If you had a counter in `_app`, and you are at the `/` route, and navigate to `/blog`, with `router.replace`, the pages doesn't unload, instead, Next.js renders the contents of `/blog` and keeps the counter state in `_app`. If you used `window.location.replace`, the page unloads, and you lose the `_app` counter state.

That being said, for outbound linking, it is a NOOP, meaning it doesn't matter. It is in the hands of the other page to redirect back to your page. People often include a query param `?return_url=` which handles that, but the "totally another website" is in charge of handling that, 100%. 

Since `replace` removes the current entry in the history for the new one, you'll not be able to just do `window.location.back()` from the "totally another website".

**Reply by Dann1y** [reply]
Thank you for your help :)
